### PR TITLE
Azure/app configuration public access

### PR DESCRIPTION
**Pull request by alphadev4**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -1038,7 +1038,11 @@ module.exports = {
         'amsDiagnosticLogsEnabled'      : require(__dirname + '/plugins/azure/mediaServices/amsDiagnosticLogsEnabled.js'),
         'amsPublicAccessDisabled'       : require(__dirname + '/plugins/azure/mediaServices/amsPublicAccessDisabled.js'),
         'amsManagedIdentityEnabled'     : require(__dirname + '/plugins/azure/mediaServices/amsManagedIdentityEnabled.js'),
-        'amsClassicApiDisabled'         : require(__dirname + '/plugins/azure/mediaServices/amsClassicApiDisabled.js')
+        'amsClassicApiDisabled'         : require(__dirname + '/plugins/azure/mediaServices/amsClassicApiDisabled.js'),
+    
+        'appConfigurationManagedIdentity': require(__dirname + '/plugins/azure/appConfigurations/appConfigurationManagedIdentity.js'),
```
Comment (mehakseedat63) [inline exports.js:1043]:
```suggestion
        'appConfigurationManagedIdentity': require(__dirname + '/plugins/azure/appConfigurations/appConfigurationManagedIdentity.js'),
```
can we change it to appConfigManagedIdentity?

File: plugins/azure/appConfigurations/appConfigurationManagedIdentity.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'App Configurations Managed Identity',
+    category: 'App Configuration',
+    domain: 'Developer Tools',
+    description: 'Ensures that Azure App Configurations have managed identity enabled.',
+    more_info: 'Enabling managed identities eliminate the need for developers having to manage credentials by providing an identity for the Azure resource in Azure AD and using it to obtain Azure Active Directory (Azure AD) tokens.',
+    link: 'https://learn.microsoft.com/en-us/azure/azure-app-configuration/howto-integrate-azure-managed-service-identity?tabs=core6x&pivots=framework-dotnet',
```
Comment (mehakseedat63) [inline plugins/azure/appConfigurations/appConfigurationManagedIdentity.js:10]:
```suggestion
    link: 'https://learn.microsoft.com/en-us/azure/azure-app-configuration/overview-managed-identity',
```

File: plugins/azure/appConfigurations/appConfigurationManagedIdentity.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            if (!appConfigurations.data.length) {
+                helpers.addResult(results, 0, 'No existing App Configurations found', location);
+                return rcb();
+            }
+
+            for (let appConfiguration of appConfigurations.data) {
+                if (!appConfiguration.id) continue;
+                var identityType = appConfiguration.identity && appConfiguration.identity.type? appConfiguration.identity.type : null;
+
+                if (identityType && (identityType.includes('systemassigned') || identityType.includes('userassigned'))) {
```
Comment (mehakseedat63) [inline plugins/azure/appConfigurations/appConfigurationManagedIdentity.js:39]:
```suggestion
                if (identityType && (identityType.toLowerCase().includes('systemassigned') || identityType.toLowerCase().includes('userassigned'))) {
```
is it necessary to check for identity type?

File: plugins/azure/appConfigurations/appConfigurationManagedIdentity.spec.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+          "encryption": {
+            "keyVaultProperties": null
+          },
+          "privateEndpointConnections": null,
+          "publicNetworkAccess": "Disabled",
+          "disableLocalAuth": false,
+          "softDeleteRetentionInDays": 0,
+          "enablePurgeProtection": false
+        },
+        "id": "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourceGroups/meerab-rg/providers/Microsoft.AppConfiguration/configurationStores/meerab-test-rg",
```
Comment (mehakseedat63) [inline plugins/azure/appConfigurations/appConfigurationManagedIdentity.spec.js:21]:
```suggestion
        "id": "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourceGroups/meerab-rg/providers/Microsoft.AppConfiguration/configurationStores/meerab-test-rg",
```
change subscription id to some random value

File: plugins/azure/appConfigurations/appConfigurationManagedIdentity.spec.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
```
Comment (mehakseedat63) [inline plugins/azure/appConfigurations/appConfigurationManagedIdentity.spec.js:41]:
```suggestion
        "id": "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourceGroups/meerab-rg/providers/Microsoft.AppConfiguration/configurationStores/meerab-test-rg",
```
change subscription id to a random value

File: plugins/azure/appConfigurations/appConfigurationManagedIdentity.spec.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        "tags": {},
+        "identity": {
+            "type": "systemassigned,userassigned",
+            "principalId": "dc03d47d-e6df-491f-aebe-50a93412a890",
+            "tenantId": "d207c7bd-fcb1-4dd3-855a-cfd2f9b651e8",
+            "userAssignedIdentities": {
+              "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourcegroups/meerab-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testmeerab": {
+                "PrincipalId": "1d34c2cd-bd53-487d-b3a9-6064465497c9",
+                "ClientId": "2071caa1-3668-4de3-babc-155cfe3e38e5"
+              }
```
Comment (mehakseedat63) [inline plugins/azure/appConfigurations/appConfigurationManagedIdentity.spec.js:52]:
```suggestion
            "principalId": "dc03d47d-e6df-491f-aebe-50a93412a890",
            "tenantId": "d207c7bd-fcb1-4dd3-855a-cfd2f9b651e8",
            "userAssignedIdentities": {
              "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourcegroups/meerab-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/testmeerab": {
                "PrincipalId": "1d34c2cd-bd53-487d-b3a9-6064465497c9",
                "ClientId": "2071caa1-3668-4de3-babc-155cfe3e38e5"
              }
```
make sure we're using dummy values here

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: plugins/azure/appConfigurations/appConfigurationPublicAccess.spec.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        "endpoint": "https://meerab-test-rg.azconfig.io",
+        "encryption": {
+        "keyVaultProperties": null
+        },
+        "privateEndpointConnections": null,
+        "publicNetworkAccess": "Enabled",
+        "disableLocalAuth": false,
+        "softDeleteRetentionInDays": 0,
+        "enablePurgeProtection": false,
+        "id": "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourceGroups/meerab-rg/providers/Microsoft.AppConfiguration/configurationStores/meerab-test-rg",
```
Comment (mehakseedat63) [inline plugins/azure/appConfigurations/appConfigurationPublicAccess.spec.js:19]:
```suggestion
        "id": "/subscriptions/26a1a07e-06dd-4892-92c9-e4996b0fc546/resourceGroups/meerab-rg/providers/Microsoft.AppConfiguration/configurationStores/meerab-test-rg",
```
change these

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: exports.js [hunk 8]
```diff
@@ -1038,7 +1038,11 @@ module.exports = {
         'amsDiagnosticLogsEnabled'      : require(__dirname + '/plugins/azure/mediaServices/amsDiagnosticLogsEnabled.js'),
         'amsPublicAccessDisabled'       : require(__dirname + '/plugins/azure/mediaServices/amsPublicAccessDisabled.js'),
         'amsManagedIdentityEnabled'     : require(__dirname + '/plugins/azure/mediaServices/amsManagedIdentityEnabled.js'),
-        'amsClassicApiDisabled'         : require(__dirname + '/plugins/azure/mediaServices/amsClassicApiDisabled.js')
+        'amsClassicApiDisabled'         : require(__dirname + '/plugins/azure/mediaServices/amsClassicApiDisabled.js'),
+    
+        'appConfigManagedIdentity'      : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationManagedIdentity.js'),
```
Comment (alphadev4) [inline exports.js:1043]:
```suggestion
        'appConfigManagedIdentity'      : require(__dirname + '/plugins/azure/appConfigurations/appConfigManagedIdentity.js'),
```

**Review by mehakseedat63 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

File: exports.js [hunk 9]
```diff
@@ -1054,7 +1054,7 @@ module.exports = {
         'amsContentKeyPolicy'           : require(__dirname + '/plugins/azure/mediaServices/amsContentKeyPolicy.js'),
 
         'appConfigManagedIdentity'      : require(__dirname + '/plugins/azure/appConfigurations/appConfigManagedIdentity.js'),
-        'appConfigurationDiagnosticLogs': require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
+        'appConfigurationPublicAccess'  : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationPublicAccess.js'),
```
Comment (alphadev4) [inline exports.js:1057]:
```suggestion
        'appConfigurationDiagnosticLogs'  : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
         'appConfigurationPublicAccess'  : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationPublicAccess.js'),
```

File: exports.js [hunk 10]
```diff
@@ -1054,7 +1054,8 @@ module.exports = {
         'amsContentKeyPolicy'           : require(__dirname + '/plugins/azure/mediaServices/amsContentKeyPolicy.js'),
 
         'appConfigManagedIdentity'      : require(__dirname + '/plugins/azure/appConfigurations/appConfigManagedIdentity.js'),
-        'appConfigurationDiagnosticLogs': require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
+        'appConfigurationDiagnosticLogs'  : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
+         'appConfigurationPublicAccess'  : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationPublicAccess.js'),
```
Comment (alphadev4) [inline exports.js:1058]:
```suggestion
        'appConfigurationDiagnosticLogs' : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
        'appConfigurationPublicAccess'  : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationPublicAccess.js'),
```

File: exports.js [hunk 11]
```diff
@@ -1054,7 +1054,8 @@ module.exports = {
         'amsContentKeyPolicy'           : require(__dirname + '/plugins/azure/mediaServices/amsContentKeyPolicy.js'),
 
         'appConfigManagedIdentity'      : require(__dirname + '/plugins/azure/appConfigurations/appConfigManagedIdentity.js'),
-        'appConfigurationDiagnosticLogs': require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
+        'appConfigurationDiagnosticLogs' : require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
```
Comment (alphadev4) [inline exports.js:1057]:
```suggestion
        'appConfigurationDiagnosticLogs': require(__dirname + '/plugins/azure/appConfigurations/appConfigurationDiagnosticLogs.js'),
```

**Pull request merged**